sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v2_external/query.sql

```diff
@@ -17,7 +17,9 @@ WITH impressions AS (
     -- but we reapply truncation to be explicit about granularity.
     TIMESTAMP_TRUNC(submission_timestamp, SECOND) AS submission_timestamp,
     request_id,
-    search_query AS telemetry_query,
+    -- Firefox allows casing and whitespace differences when matching to the
+    -- list of suggestions in RemoteSettings.
+    LTRIM(LOWER(search_query)) AS telemetry_query,
```

Comment:
@ncloudioj Can you point me to the specific Firefox code that does the matching to the RemoteSettings list? Does this transformation look similar to what's done in-client?

Reply:
Yes, Firefox does [left trimming](https://searchfox.org/mozilla-central/source/browser/components/urlbar/UrlbarProviderQuickSuggest.jsm#136) and [lowercasing](https://searchfox.org/mozilla-central/source/browser/components/urlbar/UrlbarQuickSuggest.jsm#114) before the match.